src/cargo/ops/cargo_clean.rs

```diff
@@ -220,29 +222,134 @@ fn escape_glob_path(pattern: &Path) -> CargoResult<String> {
     Ok(glob::Pattern::escape(pattern))
 }
 
-fn rm_rf_glob(pattern: &Path, config: &Config) -> CargoResult<()> {
+fn rm_rf_glob(
+    pattern: &Path,
+    config: &Config,
+    progress: &mut impl CleaningProgressBar,
```

Comment:
I think it'd be ok to change this from `impl Cleaning...` to `dyn Cleaning...` since we don't need the monomorphization aspect here per se